submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/FireFilesService.kt

```diff
@@ -6,22 +6,30 @@ import ebi.ac.uk.extended.model.ExtFile
 import ebi.ac.uk.extended.model.ExtSubmission
 import ebi.ac.uk.extended.model.FireFile
 import mu.KotlinLogging
+import uk.ac.ebi.extended.serialization.service.ExtSerializationService
 import uk.ac.ebi.extended.serialization.service.FileProcessingService
+import uk.ac.ebi.extended.serialization.service.forEachFile
 
 private val logger = KotlinLogging.logger {}
 
 class FireFilesService(
     private val fireService: FireService,
     private val fileProcessingService: FileProcessingService,
+    private val serializationService: ExtSerializationService,
 ) : FilesService {
-    override fun persistSubmissionFiles(request: FilePersistenceRequest): ExtSubmission {
-        val sub = request.submission
+    override fun persistSubmissionFiles(rqt: FilePersistenceRequest): ExtSubmission = processFiles(rqt.submission)
+    override fun cleanSubmissionFiles(sub: ExtSubmission) = cleanPreviousFiles(sub)
 
+    private fun cleanPreviousFiles(sub: ExtSubmission) {
+        fun cleanFile(file: ExtFile, index: Int) {
```

Comment:
Is this a standard we'll start using? Defining functions as variables instead of private methods?

Reply:
in this case I found then usefull as they can access the sub paramter in main function and we keep function signature key.